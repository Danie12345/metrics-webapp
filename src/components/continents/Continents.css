.countries {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 18px;
}

.countries > * {
  margin: 8px 0;
}

.select-continent {
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  background-color: rgb(139, 0, 0, 0.3);
  border: none;
  border-radius: 4px;
  padding: 2px 8px;
  width: auto;
  color: whitesmoke;
  font-family: 'Montserrat', sans-serif;
  font-weight: 600;
  font-size: 18px;
  text-align: center;
  z-index: 1;
  outline: none;
}

.select-continent option {
  color: whitesmoke;
  font-family: 'Montserrat', sans-serif;
  font-weight: 500;
  font-size: 16px;
  border: none;
  background-color: rgb(139, 0, 0);
}
