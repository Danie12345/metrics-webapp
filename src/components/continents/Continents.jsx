import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { getCountries } from '../../redux/continentCountries/continentCountries';

import store from '../../redux/configureStore';

const Continents = () => {
  const dispatch = useDispatch();

  const countries = useSelector((state) => state.countries);
  const continents = ['Africa', 'Americas', 'Asia', 'Europe', 'Oceania'];
  const [continent, setContinent] = useState('Africa');
  const continentSelect = (e) => {
    setContinent(e.target.value);
    getCountries(dispatch, store.getState, e.target.value);
  };

  useEffect(() => {
    getCountries(dispatch, store.getState, continent);
  }, []);

  return (
    <div className="countries">
      <select
        value={continent}
        onChange={(e) => continentSelect(e)}
        className="browser-default custom-select"
      >
        {
          continents.map(
            (continent) => <option value={continent} key={continent}>{continent}</option>,
          )
        }
      </select>
      <ul className="country-list">
        {countries && countries.map((country) => (
          <li className="country" key={country.name.common}>
            <NavLink to={country.name.common} className="nav-link">
              <span>{country.name.common.toUpperCase()}</span>
              <img style={{ width: '120px', height: 'auto' }} alt={`${country.name}'s flag.`} src={country.flag} />
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Continents;
