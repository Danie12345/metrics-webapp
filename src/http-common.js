import axios from 'axios';

const airPollutionURL = 'http://api.openweathermap.org/data/2.5/air_pollution';
const airPollutionKey = 'aafa8bdb6721fd796e15393dd7b376bf';
const openCageDataURL = 'https://api.opencagedata.com/geocode/v1/json';
const openCageDataKey = 'ae1b0f54d8f8479c9d08f45eeec144a2';
const getCountriesURL = 'https://restcountries.com/v3.1';

const airPollution = axios.create({
  baseURL: airPollutionURL,
  headers: {
    'Content-type': 'application/json',
  },
});

const openCageData = axios.create({
  baseURL: openCageDataURL,
  headers: {
    'Content-type': 'application/json',
  },
});

const getCountries = axios.create({
  baseURL: getCountriesURL,
  headers: {
    'Content-type': 'application/json',
  },
});

export {
  airPollution, airPollutionKey, openCageData, openCageDataKey, getCountries,
};
