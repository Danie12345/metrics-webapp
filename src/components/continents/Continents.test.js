import {
  render, screen, waitFor, fireEvent, act,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import Continents from './Continents';
import store from '../../redux/configureStore';
import { getCountries } from '../../http-common';
import { BrowserRouter } from "react-router-dom";


jest.mock('../../http-common');

const fetch = async () => {
  const state = {
    data: [
      {
        name: {common: 'Zimbabwe'},
        region: 'Africa',
        coords: [12, 42],
        population: 100000,
        flags: {svg: 'some svg link'},
      }
    ],
  };
  await getCountries.get.mockResolvedValue(state);
}

describe('Continents Component', () => {
  beforeEach(async () => {
    fetch();
  });

  afterEach(() => {
    act(() => store.dispatch({
      type: 'metrics/continent/SET_CONTINENT',
      payload: 'Africa',
    }));
  });

  it('renders correctly', async () => {
    render(<Provider store={store}><BrowserRouter><Continents /></BrowserRouter></Provider>);
    await waitFor(() => {
      expect(screen.getAllByText('Africa').length).toBeGreaterThan(0);
    });
  });

  it('maintains the snapshots between renders', async () => {
    const tree = render(<Provider store={store}><BrowserRouter><Continents /></BrowserRouter></Provider>);
    await expect(tree).toMatchSnapshot();
  });
});
