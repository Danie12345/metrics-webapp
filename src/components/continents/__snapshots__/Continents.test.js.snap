// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Continents Component maintains the snapshots between renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="countries"
      >
        <img
          alt="Background world."
          src="world.png"
          style="width: 100%; position: fixed; top: 35%; z-index: 1;"
        />
        <select
          class="select-continent"
        >
          <option
            value="Africa"
          >
            Africa
          </option>
          <option
            value="Americas"
          >
            Americas
          </option>
          <option
            value="Asia"
          >
            Asia
          </option>
          <option
            value="Europe"
          >
            Europe
          </option>
          <option
            value="Oceania"
          >
            Oceania
          </option>
        </select>
        <ul
          class="country-list"
        >
          <li
            class="country"
          >
            <a
              class="nav-link"
              href="/continents/country/Zimbabwe"
            >
              <img
                alt="Zimbabwe's flag."
                src="some svg link"
              />
              <div
                class="country-info-small"
              >
                <span>
                  ZIMBABWE
                </span>
                <span>
                  Pop: 100000
                </span>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="countries"
    >
      <img
        alt="Background world."
        src="world.png"
        style="width: 100%; position: fixed; top: 35%; z-index: 1;"
      />
      <select
        class="select-continent"
      >
        <option
          value="Africa"
        >
          Africa
        </option>
        <option
          value="Americas"
        >
          Americas
        </option>
        <option
          value="Asia"
        >
          Asia
        </option>
        <option
          value="Europe"
        >
          Europe
        </option>
        <option
          value="Oceania"
        >
          Oceania
        </option>
      </select>
      <ul
        class="country-list"
      >
        <li
          class="country"
        >
          <a
            class="nav-link"
            href="/continents/country/Zimbabwe"
          >
            <img
              alt="Zimbabwe's flag."
              src="some svg link"
            />
            <div
              class="country-info-small"
            >
              <span>
                ZIMBABWE
              </span>
              <span>
                Pop: 100000
              </span>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
